### PR TITLE
feat(Modal): modal improvements, add stories for main Modal and modal's layouts

### PR DESCRIPTION
**Pull request by YossiSaadi**
- This PR includes stories for new Modal (hidden for now)
- This PR introduces a few feats as well:
	- New responsiveness logic
	- Remove footer shadow from Media layout
	- Remove scrollable abilities from Media layout
	- Remove pre-defined height from Media layout

https://monday.monday.com/boards/3532714909/pulses/7920875908
https://monday.monday.com/boards/3532714909/pulses/7719052213

### CODE REVIEW

File: packages/core/src/components/ModalNew/layouts/ModalSideBySideLayout/__stories__/assets/index.ts
Comment (talkor) [inline packages/core/src/components/ModalNew/layouts/ModalSideBySideLayout/__stories__/assets/index.ts:1]: Why you need it?

File: packages/core/src/components/ModalNew/layouts/ModalMediaLayout/__stories__/assets/index.ts
Comment (talkor) [inline packages/core/src/components/ModalNew/layouts/ModalMediaLayout/__stories__/assets/index.ts:1]: why needed?

File: packages/core/.storybook/preview-head.html [hunk 1]
```diff
@@ -41,15 +41,19 @@
...
+    /* prevent Show Code from rendering above full-screen components (such as Modal's overlay) */
+    z-index: unset;
+    background: unset;
+  }
 
   .sbdocs {
     color: var(--sb-primary-text-color);
   }
 
-  .sbdocs p a,
```
Comment (talkor) [inline packages/core/.storybook/preview-head.html:52]: why remove?

File: packages/core/src/components/ModalNew/Modal/__stories__/Modal.stories.helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  size?: "small" | "medium" | "large";
+  children: React.ReactNode;
+}) => {
+  const isAfterFirstRender = useAfterFirstRender();
+  return (
+    <div
+      className={cx(styles.preview, getStyle(styles, size))}
+      // workaround to prevent modal from autofocusing on page load
+      // autofocus would work as expected when modal closes and reopens
+      {...(!isAfterFirstRender.current && isDocsView && { "data-no-autofocus": true })}
```
Comment (talkor) [inline packages/core/src/components/ModalNew/Modal/__stories__/Modal.stories.helpers.tsx:27]: Nice solution

File: packages/core/src/components/ModalNew/layouts/ModalBasicLayout/__stories__/ModalBasicLayout.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+    const [showLarge, setShowLarge] = useState(false);
+
+    return (
+      <>
+        <Flex gap="large" style={{ paddingBlock: 40 }}>
+          <Button onClick={() => setShowSmall(true)}>Small</Button>
+          <Button onClick={() => setShowMedium(true)}>Medium</Button>
+          <Button onClick={() => setShowLarge(true)}>Large</Button>
+        </Flex>
+        {createPortal(
```
Comment (talkor) [inline packages/core/src/components/ModalNew/layouts/ModalBasicLayout/__stories__/ModalBasicLayout.stories.tsx:87]: Why aren't we using a portal in the Modal?

**Review by talkor — APPROVED** [review]

File: packages/core/src/components/ModalNew/layouts/ModalSideBySideLayout/__stories__/assets/index.ts
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/layouts/ModalSideBySideLayout/__stories__/assets/index.ts:1]: removed

File: packages/core/src/components/ModalNew/layouts/ModalMediaLayout/__stories__/assets/index.ts
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/layouts/ModalMediaLayout/__stories__/assets/index.ts:1]: removed

File: packages/core/.storybook/preview-head.html (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/.storybook/preview-head.html]: accidentally removed, retrieved, nice catch thanks!

File: packages/core/src/components/ModalNew/Modal/__stories__/Modal.stories.helpers.tsx (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/Modal/__stories__/Modal.stories.helpers.tsx:27]: and it's not even GPT 🤣

File: packages/core/src/components/ModalNew/layouts/ModalBasicLayout/__stories__/ModalBasicLayout.stories.tsx (hunk 3 above)
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/layouts/ModalBasicLayout/__stories__/ModalBasicLayout.stories.tsx:87]: will add it, created a task

**Pull request merged**